### PR TITLE
fulfill disconnect interface

### PR DESCRIPTION
**Pull request by turbocrime**
fulfill the disconnect interface published in`@penumbra-zone/client@9`

### CODE REVIEW

File: apps/extension/src/senders/approve.ts
Comment (turbocrime) [inline apps/extension/src/senders/approve.ts:1]: some mild reorganization

File: apps/extension/src/storage/origin.ts
Comment (turbocrime) [inline apps/extension/src/storage/origin.ts:1]: storage utilities moved to storage dir

File: apps/extension/src/storage/types.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export interface LocalStorageState {
   frontendUrl: string | undefined;
   passwordKeyPrint: KeyPrintJson | undefined;
   fullSyncHeight: number | undefined;
+  // TODO: It's likely that an array is not the best data structure for this in storage. Should revisit later.
```
Comment (turbocrime) [inline apps/extension/src/storage/types.ts:25]: moved this comment to the storage types definition

File: apps/extension/src/listeners/message-prax-disconnect.ts
Comment (turbocrime) [inline apps/extension/src/listeners/message-prax-disconnect.ts:1]: the new handler

File: apps/extension/src/senders/disconnect.ts
Comment (turbocrime) [inline apps/extension/src/senders/disconnect.ts:1]: disconnect implementation

File: apps/extension/public/manifest.json [hunk 2]
```diff
@@ -41,7 +40,7 @@
   },
   "web_accessible_resources": [
     {
-      "resources": ["manifest.json"],
+      "resources": ["manifest.json", "favicon/*"],
```
Comment (turbocrime) [inline apps/extension/public/manifest.json:43]: expose the extension icon so dapps may use it

File: apps/extension/src/content-scripts/injected-penumbra-global.ts
Comment (turbocrime) [inline apps/extension/src/content-scripts/injected-penumbra-global.ts:1]: injection is refactored into a class to more clearly and explicitly manage state.

the interface is async. so to provide the synchronous `isConnected` state querying method, the state of the three possible actions are tracked, and resolved in the private function `state`.

as expected, `isConnected() => true` would indicate an active connection.

more information on the interface is available in client package documentation, but what it comes down to is:

- if the page has called `disconnect`, the connection state is false.
- if the page has called `request` but the request failed, the connection state is false.
- if no action has been attempted, the connection state is undefined.
- if connection has succeeded, the connection state is true.

**Review by grod220 — APPROVED** [review]

**Comment by grod220** [comment]
Going to take this PR over, rebase, and merge so we can cut a release given the proving key update

**Comment by turbocrime** [comment]
thanks

**Pull request merged**